mobilecoind/src/service.rs

```diff
@@ -956,12 +957,80 @@ impl<T: BlockchainConnection + UserTxConnection + 'static> ServiceApi<T> {
                 Some("receipt.tombstone".to_string()),
             ));
         }
+
         // Check if the hash landed in the ledger.
         let mut hash_bytes = [0u8; 32];
         hash_bytes.copy_from_slice(&request.get_receipt().tx_out_hash);
 
         match self.ledger_db.get_tx_out_index_by_hash(&hash_bytes) {
             Ok(_) => {
+                // If a monitor ID was given then validate the confirmation number
+                match request.get_monitor_id().len() {
+                    0 => { /* no monitor ID given */ }
+                    32 => {
+                        let sender_monitor_id =
+                            MonitorId::try_from(&request.monitor_id).map_err(|err| {
```

Comment:
This is not the sender id (since this call is `get_tx_status_as_receiver_impl`).